src/components/directory-item-selector.js

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import TreeViewFinder from '@gridsuite/commons-ui/lib/components/TreeViewFinder';
+import PropTypes from 'prop-types';
+import { fetchDirectoryContent, fetchRootFolders } from '../utils/rest-api';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    getIconFor,
+    elementType,
+} from '@gridsuite/commons-ui/lib/utils/elementType';
+
+const useStyles = makeStyles((theme) => ({
+    icon: {
+        marginRight: theme.spacing(1),
+        width: '18px',
+        height: '18px',
+    },
+}));
+
+const DirectoryItemSelector = (props) => {
+    const [data, setData] = useState([]);
+    const nodeMap = useRef({});
+    const classes = useStyles();
+
+    const contentFilter = useCallback(
+        () => new Set([elementType.DIRECTORY, ...props.types]),
+        [props.types]
+    );
+
+    const directory2Tree = useCallback(
+        (newData) => {
+            const newNode = {
+                id: newData.elementUuid,
+                name: newData.elementName,
+                icon: getIconFor(newData.type, classes.icon),
+                children:
+                    newData.type === elementType.DIRECTORY
+                        ? [{ id: '', name: '' }]
```

Comment:
I don't like this virtual/phantom child, we should make changes to TreeFinder component if there is a problem with empty directories which could be considered as leaf and then become selectable.